### DISCUSSION TITLE
Best Practices for Self-Hosting searX

### DISCUSSION THREAD

**Discussion by ghost** [searx/searx#2370] [General]
Hi, 

I would like to thank all the devs who has made this possible. Please keep up the good work.
I apologize upfront if it is too basic of a query.

I am using searx via manual installation steps, in particular using _virtualenv._  
Operating system: Ubuntu 20.04

The trouble I am having is, after using it for around 2 weeks, now mostly all search engines are getting timeout. To name a few; qwant, searchpage, google and even bing sometimes. I know basics of computer networking but I am not well-versed in it so I am not able to conclude what might be causing this. 

The installation is just virtualenv, no uwsgi, nginx, search proxy. I wanted to understand;
-  If I follow any of these routes (especially search proxy), will it results in less failure rates?
-  How does the VPN affects the results.
- Does browser fingerprinting increases the failure rate? (Extensions I am using with Firefox: uBlock Origin, Privacy Badger, HTTPs everywhere, ClearURLs)

If anyone can please help understanding these or you can direct me towards any resource that would really help. My main objective is to bring in my close friends and family, I can help them setup but I myself has some doubts & having trouble debugging things in my self-hosted engine.

Any help / advise would be highly appreciated.

**Comment by return42** [comment]
> The trouble I am having is, after using it for around 2 weeks, now mostly all search engines are getting timeout.

You should protect you instance by using filtron: https://searx.github.io/searx/admin/filtron.html

> My main objective is to bring in my close friends and family, 

For better privacy you should also consider to use [morty](https://searx.github.io/searx/utils/morty.sh.html) as image proxy.

> Best Practices for Self-Hosting searX

Its recommended to do a [docker installation](https://searx.github.io/searx/admin/installation-docker.html#installation-docker) or  to use the [installation scripts](https://searx.github.io/searx/admin/installation.html#installation-scripts).  Both methods include a _application firewall_ [filtron](https://searx.github.io/searx/admin/filtron.html) and a _result proxy_ [morty](https://searx.github.io/searx/utils/morty.sh.html) installation.
